cache/cache_deleter.go

```diff
@@ -0,0 +1,51 @@
+// Package cache provides functionalities around the cache
+package cache
+
+import (
+	"github.com/buildpacks/imgutil"
+	"github.com/pkg/errors"
+
+	"github.com/buildpacks/lifecycle/log"
+)
+
+// ImageDeleter defines the methods available to delete and compare cached images
+type ImageDeleter interface {
+	DeleteImage(image imgutil.Image)
+	OldImageIsTheSameAsTheNewImage(oldImage imgutil.Image, newImage imgutil.Image) bool
```

Comment:
In this interface is the "old" and "new" important or could this method be given a shorter name like `ImagesEq` or `Equals` ?